zds/tutorialv2/models/database.py

```diff
@@ -1515,4 +1516,27 @@ def transfer_paternity_receiver(sender, instance, **kwargs):
     PublishedContent.objects.transfer_paternity(instance, external)
 
 
+class Clap(models.Model):
+    class Meta:
+        verbose_name = "Clap"
+        verbose_name_plural = "Claps"
+
+    content = models.ForeignKey(
+        PublishableContent,
+        verbose_name="Contenu",
+        on_delete=models.CASCADE,
+        related_name="claps",
+        db_index=True,
+    )
+    user = models.ForeignKey(User, db_index=True, on_delete=models.SET_NULL, null=True)
+    hash_ip_address = models.CharField(db_index=True, max_length=40)
+    claps_count = models.IntegerField()
+    date = models.DateTimeField(auto_now_add=True)
+
+    @staticmethod
+    def hash_ip(request) -> str:
+        ip_address = request.META.get("HTTP_X_FORWARDED_FOR") or request.META.get("REMOTE_ADDR")
+        return str(hashlib.sha256(bytes(settings.SECRET_KEY + ip_address, "utf-8")).hexdigest())
```

Comment:
`hexdigest` retourne déjà un objet `str` :)
```suggestion
        return hashlib.sha256(bytes(settings.SECRET_KEY + ip_address, "utf-8")).hexdigest()
```